narwhals/_pandas_like/series.py

```diff
@@ -549,6 +549,14 @@ def millisecond(self) -> PandasSeries:
     def microsecond(self) -> PandasSeries:
         return self._series._from_series(self._series._series.dt.microsecond)
 
+    def nanosecond(self) -> PandasSeries:
+        return self._series._from_series(
+            (
+                (self._series._series.dt.microsecond * 1e3).astype(int)
```

Comment:
```suggestion
                self._series._series.dt.microsecond * 1_000
```
like this we avoid casting to float and back